pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -760,31 +759,37 @@ impl DeltaLayerInner {
         &self.layer_lsn_range
     }
 
-    /// Returns nested result following Result<Result<_, OpErr>, Critical>:
-    /// - inner has the success or transient failure
-    /// - outer has the permanent failure
     pub(super) async fn load(
         path: &Utf8Path,
         summary: Option<Summary>,
         max_vectored_read_bytes: Option<MaxVectoredReadBytes>,
         ctx: &RequestContext,
-    ) -> Result<Result<Self, anyhow::Error>, anyhow::Error> {
+    ) -> Result<Self, LoadError> {
         let file = match VirtualFile::open(path, ctx).await {
             Ok(file) => file,
-            Err(e) => return Ok(Err(anyhow::Error::new(e).context("open layer file"))),
+            Err(e) => {
+                return Err(LoadError::Io(
+                    anyhow::Error::new(e).context("open layer file"),
+                ));
```

Comment:
This could be `LoadError::Open`, second one could be `LoadError::Read`, third could be `LoadError::Corruption`, none need an anyhow...